rules/prefer-object-from-entries.js

```diff
@@ -0,0 +1,274 @@
+		}
+
+		let shouldAddParentheses = false;
+		if (parameters.length === 1) {
+			const arrowToken = sourceCode.getTokenAfter(firstParameter, isArrowToken);
+			const tokenBeforeArrowToken = sourceCode.getTokenBefore(arrowToken);
+
+			if (!isClosingParenToken(tokenBeforeArrowToken)) {
+				shouldAddParentheses = true;
+			}
+		}
+
+		yield fixer.replaceText(firstParameter, shouldAddParentheses ? '()' : '');
+	}
+
+	function getKeyValueText() {
+		const {key, value} = property;
+		let keyText = getParenthesizedText(key, sourceCode);
+		const valueText = getParenthesizedText(value, sourceCode);
+
+		if (!property.computed && key.type === 'Identifier') {
+			keyText = `'${keyText}'`;
+		}
+
+		return {keyText, valueText};
+	}
+
+	function * replaceFunctionBody(fixer) {
+		const functionBody = node.arguments[0].body;
+		const {keyText, valueText} = getKeyValueText();
+		yield fixer.replaceText(functionBody, `[${keyText}, ${valueText}]`);
+
+		for (const parentheses of getParentheses(functionBody, sourceCode)) {
+			yield fixer.remove(parentheses);
+		}
+	}
+
+	return function * (fixer) {
+		// Wrap `array.reduce()` with `Object.fromEntries()`,
+		yield fixer.insertTextBefore(node, 'Object.fromEntries(');
+		yield fixer.insertTextAfter(node, ')');
+
+		// Switch `.reduce` to `.map`
+		yield fixer.replaceText(node.callee.property, 'map');
+
+		// Remove empty object
+		yield removeInitObject(fixer);
+
+		// Remove the first parameter
+		yield * removeFirstParameter(fixer);
+
+		// Replace function body
+		yield * replaceFunctionBody(fixer);
+	};
+}
+
+/** @param {import('eslint').Rule.RuleContext} context */
+function create(context) {
+	const {functions: configFunctions} = {
+		functions: [],
+		...context.options[0],
+	};
+	const functions = [...configFunctions, ...lodashFromPairsFunctions];
+	const sourceCode = context.getSourceCode();
+	const listeners = {};
+	const arrayReduce = new Map();
+
+	for (const {selector, test, getProperty} of fixableArrayReduceCases) {
+		listeners[selector] = function (node) {
+			// If this listener exit without adding fix, the `arrayReduceWithEmptyObject` listener
+			// should still add it into the `arrayReduce` map, to be safer, add it here too
+			arrayReduce.set(node, undefined);
+
+			const [callbackFunction] = node.arguments;
+			if (!test(callbackFunction)) {
+				return;
+			}
+
+			const [firstParameter] = callbackFunction.params;
+			const variables = context.getDeclaredVariables(callbackFunction);
+			const firstParameterVariable = variables.find(variable => variable.identifiers.length === 1 && variable.identifiers[0] === firstParameter);
+			if (!firstParameterVariable || firstParameterVariable.references.length !== 1) {
+				return;
+			}
+
+			arrayReduce.set(
+				node,
+				// The fix function
+				fixReduceAssignOrSpread({
+					sourceCode,
+					node,
+					property: getProperty(callbackFunction),
+				}),
+			);
+		};
+	}
+
+	listeners[arrayReduceWithEmptyObject] = function (node) {
```

Comment:
```suggestion
	listeners[arrayReduceWithEmptyObject] = node => {
```

Same with the other ones.